meteor/client/ui/Settings/ShowStyle/VariantSettings.tsx

```diff
@@ -103,21 +291,82 @@ export const ShowStyleVariantsSettings = withTranslation()(
 			})
 		}
 
-		renderShowStyleVariants() {
+		VariantItem = ({ index, showStyleVariant, moveVariantHandler }) => {
```

Comment:
This is a unusual design pattern - Is there a particular reason why this needs to be a member of `ShowStyleVariantsSettings` class? It seems like it could just accept some additional props and a callback and it wouldn't have to be a member. Also, extracting this function from a class member into a separate file & component would help with reducing the size of the `VariantSettings` file.

Reply:
@RasmusAlbrektsen

Reply:
I agree that this mixing of style (class vs function react component) is unexpected. In my opinion a file should only contain either class or functional components.
In addition, this has lead to some of the newly added callbacks which are passed into child components to be written using `useCallback` and some to be members of the parent class

Reply:
This can largely be attributed to my unfamiliarity to React and how the flows work here. The intention was to make it easy to later retract from the file and divide into separate files to easier make it generic for a later task down the road. But again, it is mostly because i had issues with figuring out how to make it generic at this point in time. 
I will gladly take any advice on how this could be done better, i'm still very much learning when it comes to how the UI is put together!